### DISCUSSION TITLE
RuntimeError while trying to run drive cycle in experiment with initial SOC

### DISCUSSION THREAD

**Discussion by Mrzhang-hub** [pybamm-team/PyBaMM#2489] [Q&A]
Hi! I am trying to run a drive cycle in an experiment but it keeps failing. Part of the code as follows:

```
options = {"lithium plating":"irreversible","thermal": "lumped"}
model = pybamm.lithium_ion.SPMe(options=options)
param = pybamm.ParameterValues('OKane2022')
soc_init = 0.2
drive_cycle_current = pd.read_csv("StandardCycle/UDDS.csv", comment="#", header=None).to_numpy()
exp = pybamm.Experiment([("Run UDDS_A (A)")],drive_cycles={"UDDS_A": drive_cycle_current})
sim = pybamm.Simulation(model=model, parameter_values=param, experiment=exp)
sol = sim.solve(initial_soc=soc_init)

```

It gives the error:

```
RuntimeError: .../casadi/core/slice.cpp:83: Assertion "stop<=len" failed:
Slice (start=0, stop=20, step=1) out of bounds with supplied length of 1
```

But when I remove the lithium plating option, the code works well.  

Why? Can you help me handle this? @rtimms  @tinosulzer

**Comment by RuiheLi** [comment]
Hi, I think this error you showed is related to the solver, which is quite difficult to debug. PyBaMM tries to give as much information as possible such as [exceptions](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/expression_tree/exceptions.py). However, there are always some other problems out of these exceptions. That is the tricky part of simulation which I think it is quite generous. However, based on my personal experience, there are still something worth checking:
1. voltage, such as terminal voltage, negative/positive electrode potential, electrolyte potential, etc.
2. concentration, especially particle surface concentration and electrolyte concentration
3. porosity, especially negative electrode porosity

Other things to try are to reduce dt_max, try another solver, etc.. Also, I found that reducing the time to exactly when the model fails can probably give you more information. To do that, you can divide your whole experiments / cycles into smaller pieces and use the `set_initial_conditions_from`  function in PyBaMM to run step by step.  

Also, it is worth reading through the examples that PyBaMM provides [here](https://github.com/pybamm-team/PyBaMM/tree/develop/examples), sometimes you may find some useful scripts/functions. 

I currently have not found any shortcuts in debugging or avoiding errors. Though I wish I have~ lol.
